ui/src/shared/components/AnnotationTooltip.tsx

```diff
@@ -1,140 +1,76 @@
-  time: string
+  time: number
 }
 
-const TimeStamp = ({time}: TimeStampProps): JSX.Element => (
+const TimeStamp: SFC<TimeStampProps> = ({time}) => (
   <div className="annotation-tooltip--timestamp">
-    {`${moment(+time).format('YYYY/MM/DD HH:mm:ss.SS')}`}
+    {`${moment(time).format('YYYY/MM/DD HH:mm:ss.SS')}`}
   </div>
 )
 
 interface AnnotationState {
-  isDragging: boolean
-  isMouseOver: boolean
-}
-
-interface Span {
-  spanCenter: number
-  tooltipLeft: number
-  spanWidth: number
-}
-
-interface State {
-  annotation: AnnotationInterface
+  isDragging: string | boolean
+  isMouseOver: string | boolean
 }
 
 interface Props {
-  isEditing: boolean
-  annotation: AnnotationInterface
-  timestamp: string
-  onMouseLeave: (e: MouseEvent<HTMLDivElement>) => {}
+  annotation: Annotation
+  timestamp: number
+  onMouseLeave: (e: MouseEvent<HTMLDivElement>) => void
   annotationState: AnnotationState
-  deleteAnnotationAsync: (a: AnnotationInterface) => void
-  updateAnnotationAsync: (a: AnnotationInterface) => void
-  span: Span
+  onSetEditingAnnotation: typeof setEditingAnnotation
 }
 
-@ErrorHandling
-class AnnotationTooltip extends Component<Props, State> {
-  public state = {
-    annotation: this.props.annotation,
-  }
-
-  public componentWillReceiveProps(nextProps: Props) {
-    const {annotation} = nextProps
-    this.setState({annotation})
-  }
-
-  public render() {
-    const {annotation} = this.state
-    const {
-      onMouseLeave,
-      timestamp,
-      annotationState: {isDragging, isMouseOver},
-      isEditing,
-      span,
-    } = this.props
-
-    const tooltipClass = classnames('annotation-tooltip', {
-      hidden: !(isDragging || isMouseOver),
-      'annotation-span-tooltip': !!span,
-    })
-
-    return (
-      <div
-        id={`tooltip-${annotation.id}`}
-        onMouseLeave={onMouseLeave}
-        className={tooltipClass}
-        style={
-          span
-            ? {left: `${span.tooltipLeft}px`, minWidth: `${span.spanWidth}px`}
-            : {}
-        }
-      >
-        {isDragging ? (
-          <TimeStamp time={timestamp} />
-        ) : (
-          <div className="annotation-tooltip--items">
-            {isEditing ? (
-              <div>
-                <AnnotationInput
-                  value={annotation.text}
-                  onChangeInput={this.handleChangeInput('text')}
-                  onConfirmUpdate={this.handleConfirmUpdate}
-                  onRejectUpdate={this.handleRejectUpdate}
-                />
-                <button
-                  className="annotation-tooltip--delete"
-                  onClick={this.handleDelete}
-                  title="Delete this Annotation"
-                >
-                  <span className="icon trash" />
-                </button>
-              </div>
-            ) : (
-              <div>{annotation.text}</div>
-            )}
-            <TimeStamp time={timestamp} />
+const AnnotationTooltip: SFC<Props> = props => {
```

Comment:
typically i think SFCs are ultimately a premature optimization that we end up refactoring into Components anyway, but hopefully in this case this stays as SFC since it seems like the subtle performance boost could be useful